File: 1089-duplicate-zeros/1089-duplicate-zeros.java
class Solution {
    public void duplicateZeros(int[] arr) {
        for(int i = 0; i<arr.length; i++){
            if(arr[i] == 0){
                arr = rightShift(arr, i);
                i++;
            }
        }
    }

    public int[] rightShift(int[] arr, int idx){
        if(idx < arr.length){
            for(int i = arr.length-1; i>idx; i--){
                arr[i] = arr[i-1];
            }
        }

        return arr;
    }
}

File: 1089-duplicate-zeros/1089-duplicate-zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        
        for i in range(len(arr)-1, -1, -1):

            if arr[i] == 0:
                for j in range(len(arr)-1, i, -1):
                    arr[j] = arr[j-1]
        
        """
        Do not return anything, modify arr in-place instead.
        """
        